src/benchmarks/micro/libraries/System.Formats.Tar/Perf.TarFile.cs

```diff
@@ -50,21 +51,49 @@ public void CreateFromDirectory_Path()
             File.Delete(_outputTarFilePath);
         }
 
+        [Benchmark]
+        public async Task CreateFromDirectory_Path_Async()
+        {
+            await TarFile.CreateFromDirectoryAsync(sourceDirectoryName: _inputDirPath, destinationFileName: _outputTarFilePath, includeBaseDirectory: false);
+            File.Delete(_outputTarFilePath);
+        }
+
         [Benchmark]
         public void ExtractToDirectory_Path() => TarFile.ExtractToDirectory(sourceFileName: _inputTarFilePath, destinationDirectoryName: _outputDirPath, overwriteFiles: true);
 
+        [Benchmark]
+        public Task ExtractToDirectory_Path_Async() => TarFile.ExtractToDirectoryAsync(sourceFileName: _inputTarFilePath, destinationDirectoryName: _outputDirPath, overwriteFiles: true);
+
         [Benchmark]
         public void CreateFromDirectory_Stream()
         {
             using MemoryStream ms = new MemoryStream();
             TarFile.CreateFromDirectory(sourceDirectoryName: _inputDirPath, destination: ms, includeBaseDirectory: false);
         }
 
+        [Benchmark]
+        public async Task CreateFromDirectory_Stream_Async()
+        {
+            await using (MemoryStream ms = new MemoryStream())
```

Comment:
I'm using the brackets to wrap the internal await because I want to avoid the race condition I was seeing in the tar unit tests: If you use `using` to dispose an object, then make that same method return a Task (instead of awaiting it), then the Task will most likely get processed _after_ the disposable object is disposed when leaving its context.

Reply:
this doesn't generate the same code as  this?

```c#
await using MemoryStream ms = new MemoryStream();
await TarFile.CreateFromDirectoryAsync(sourceDirectoryName: _inputDirPath, destination: ms, includeBaseDirectory: false);
```

Reply:
these produce the same code, perhaps I misunderstand --
```c#
    static async Task Main1()
    {
        await using MemoryStream ms = new MemoryStream();
        await ms.ReadAsync(null);
    }
    
    static async Task Main2()
    {
        await using (MemoryStream ms = new MemoryStream())
        {
            await ms.ReadAsync(null);
        }
    }
```

Reply:
Oh right. The `await` next to the `using` makes the method async.

The race condition I was seeing came from creating a `using MemoryStream ms = new();` (non- async), then returning the task.